app/screens/find_channels/filtered_list/filtered_list.tsx

```diff
@@ -275,24 +302,36 @@ const FilteredList = ({
         }
 
         return [...new Set(items)].slice(0, MAX_RESULTS + 1);
-    }, [archivedChannels, channelsMatchStart, channelsMatch, remoteChannels, usersMatch, usersMatchStart, locale, teammateDisplayNameSetting]);
+    }, [archivedChannels, channelsMatchStart, channelsMatch, remoteChannels, usersMatch, usersMatchStart, locale, teammateDisplayNameSetting, showThreadItem]);
 
     useEffect(() => {
         mounted.current = true;
         return () => {
             mounted.current = false;
         };
-    });
+    }, []);
+
+    const threadLabel = useMemo(
+        () => formatMessage({
+            id: 'threads',
+            defaultMessage: 'Threads',
+        }).toLowerCase(),
+        [locale],
+    );
 
     useEffect(() => {
         bounce.current = debounce(search, 500);
         bounce.current();
+        if (isCRTEnabled) {
+            const isThreadTerm = threadLabel.indexOf(term.toLowerCase()) === 0;
+            setShowThreadItem(isThreadTerm);
+        }
```

Comment:
Moving the code to the useMemo but the useMemo will need to have "term" as  dependency  and it will lead to the same number of re-renders except for the first time it matches the query but it also eliminates `setShowThreadItem` state.